html/js/product-search.js

```diff
@@ -109,10 +152,12 @@ function rank_products(products, product_preferences, use_user_product_preferenc
 	
 	if (use_user_product_preferences_for_ranking) {
 	
-		// Rank all products, and return them in 3 arrays: "yes", "no", "unknown"
+		// Rank all products
 		
 		products.sort(function(a, b) {
-			return (b.match_score - a.match_score) || (a.initial_order - b.initial_order);
+			return (b.match_score - a.match_score)	// Highest score first
+				|| ((a.match_status == "does_not_match" ? 1 : 0) - (b.match_status == "does_not_match" ? 1 : 0)) // Matching products second
```

Comment:
You should use `===` for matching in js.
Why "does_not_match" gives me 1 point instead of 0 ?

Reply:
Also is it normal that we do not consider the other status like may_not_match ?

Reply:
hum I see, it's because you reversed a bnd b order compared to above.

I would better have:
```javascript
((b.match_status == "does_not_match" ? 0 : 1) - (a.match_status == "does_not_match" ? 0 : 1)) // Matching products second
```
but it's ok.

Reply:
Ok, I'll change, and use === instead of ==